src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -425,39 +423,68 @@ public void storeLocation(Location location)
         e.printStackTrace();
       }
     }
-    
-    if(locations.size()==1)//if one location found
-    {
-        Query allLocations =
-           new Query("Location")
-                   .setFilter(new Query.FilterPredicate("user", FilterOperator.NOT_EQUAL, user));
-        PreparedQuery allLocationresults = datastore.prepare(allLocations);
-        for (Entity entity : allLocationresults.asIterable()) {
-          try {
-            double otherLongitude = Double.parseDouble(entity.getProperty("longitude").toString());
-            double otherLatitude = Double.parseDouble(entity.getProperty("latitude").toString());
-            if(Math.abs(locations.get(0).getLatitude()-otherLatitude)<3 && Math.abs(locations.get(0).getLongitude()-otherLongitude)<3)
-            {
-              String idString = entity.getKey().getName();
-              UUID id = UUID.fromString(idString);
-              String text = (String) entity.getProperty("text");
-              String otherUser = (String) entity.getProperty("user");
-              Location location = new Location(id, otherLatitude, otherLongitude, text, otherUser);
-              locations.add(location);
-            }
-
-          } catch (Exception e) {
-            System.err.println("Error reading message.");
-            System.err.println(entity.toString());
-            e.printStackTrace();
+    if(locations.size()==0 || locations.size()>1)//location does not exist or multiple invalid locations
+        return null;
+    else
+        return locations.get(0);
+  }
+  
+  /*return list of charity locations near a user location*/
+  public List<Location> getAllNearCharityLocations(Location location) {   
+    List<Location> locations = new ArrayList<>();
+    Query allLocations =
+       new Query("Location")
+               .setFilter(new Query.FilterPredicate("user", FilterOperator.NOT_EQUAL, location.getUser()));
+    PreparedQuery allLocationresults = datastore.prepare(allLocations);
+    for (Entity entity : allLocationresults.asIterable()) {
+      try {
+        double otherLongitude = Double.parseDouble(entity.getProperty("longitude").toString());
+        double otherLatitude = Double.parseDouble(entity.getProperty("latitude").toString());
+        if(Math.abs(location.getLatitude()-otherLatitude)<3 && Math.abs(location.getLongitude()-otherLongitude)<3)
```

Comment:
Suggest to factor out the hardcoded 3 into a private static final field with a descriptive name instead (ie. LOCATION_DEGREES_DIFF?). This will improve code readability.

Reply:
Thank you, I will do that